### PR TITLE
Handle compatibility branching in a compat module

### PR DESCRIPTION
**Pull request by jpadilla**
Ref #67

### CODE REVIEW

**Comment by jpadilla** [comment]
@wbolster @mark-adams would like your thoughts on this.

**Comment by jpadilla** [comment]
@wbolster @mark-adams will probably merge this if there aren't no objections and release a new version with all your awesome work.

**Comment by wbolster** [comment]
Sure, next step is to move all actual code out of `__init.py__`, and only import public API there. :-)

File: jwt/compat.py [hunk 1]
```diff
@@ -0,0 +1,59 @@
...
+try:
+    import json
+except ImportError:
+    import simplejson as json
+
+
+if sys.version_info >= (3, 0, 0):
+    unicode = str
+    basestring = str
+else:
```
Comment (wbolster) [inline jwt/compat.py]: drop this branch, it doesn't do anything

File: jwt/compat.py [hunk 2]
```diff
@@ -0,0 +1,59 @@
...
+import sys
+import hmac
+
+try:
+    import json
+except ImportError:
+    import simplejson as json
+
+
+if sys.version_info >= (3, 0, 0):
```
Comment (wbolster) [inline jwt/compat.py]: make this a PY3 constant (like six), and use it elsewhere as well

**Comment by mark-adams** [comment]
I know this is already merged, but I just wanted to say "Looks great!" :-)

File: jwt/compat.py (hunk 2 above)
Comment (jpadilla) [inline jwt/compat.py]: @wbolster thanks, borrowed what six was doing for this.

**Comment by jpadilla** [comment]
@wbolster @mark-adams thanks again.

@wbolster RE moving the actual code out of __init__, on that right now. Will open a PR in a few minutes.

**Comment by mark-adams** [comment]
Interesting note regaridng moving code out of init: A lot of this has been done in #71 :-)

**Comment by jpadilla** [comment]
@mark-adams true that, will hold off on this until then.

**Comment by jpadilla** [comment]
Just released [v0.4.1](https://github.com/progrium/pyjwt/releases/tag/0.4.1).

**Pull request merged**